### PR TITLE
Let RPC services bind to a port

### PR DESCRIPTION
**Pull request by davide125**
Spun out of #25 as requested.

Original patch: https://src.fedoraproject.org/rpms/xinetd/c/f48798436d1961de0228e92a02c1fc82ce9260f5?branch=rawhide
Related: https://bugzilla.redhat.com/show_bug.cgi?id=624800

### CODE REVIEW

**Comment by davide125** [comment]
Related issue: https://github.com/openSUSE/xinetd/issues/21

File: src/service.c [hunk 1]
```diff
@@ -156,6 +156,7 @@ static status_e activate_rpc( struct service *sp )
    socklen_t              sin_len = sizeof(tsin);
    unsigned long          vers ;
    struct service_config *scp = SVC_CONF( sp ) ;
+   uint16_t               service_port = SC_PORT( scp ) ;
```
Comment (vapier) [inline src/service.c:159]: shouldn't this leverage `SC_SPECIFIED( scp, A_PORT )` ?  and set it to 0 if port isn't specified ?

**Review by vapier — CHANGES_REQUESTED** [review]
i think the section discussing port & rpc services in man/xinetd.conf.5 needs updating too

File: src/service.c (hunk 1 above)
Comment (davide125) [reply src/service.c:159]: The usage here matches the existing logic in https://github.com/openSUSE/xinetd/blob/master/src/service.c#L236

I think both of these could probably be updated to use `SC_SPECIFIED( scp, A_PORT )` but that seems more appropriate as a separate followup change.

Comment (vapier) [reply src/service.c:159]: you want to do that as another commit in this PR ? :)

Comment (davide125) [reply src/service.c]: Here's a quick attempt.

File: src/service.c [hunk 2]
```diff
@@ -170,11 +171,16 @@ static status_e activate_rpc( struct service *sp )
    if ( SC_PROTOVAL ( scp ) == IPPROTO_TCP ) {
       M_SET ( scp->sc_xflags, SF_NOLIBWRAP );
    }
+   if ( !service_port ) {
+      service_port = 0 ;
```
Comment (vapier) [inline src/service.c:175]: this logic doesn't make sense ... the only way this `if` statement was true is if `service_port` was already `0`

i think what you want to write:
`service_port = SC_SPECIFIED( scp, A_PORT ) ? SC_PORT( scp ) : 0;`

**Comment by vapier** [comment]
thx for stickin' through my nits

**Pull request merged**